### DISCUSSION TITLE
regenerate Img/tmp on order history

### DISCUSSION THREAD

**Discussion by YoniSaadoun** [PrestaShop/PrestaShop#33443] [Ideas]
#### Describe the bug

The images product displayed in the order details are not the same as the images product.
Image product with watermark, but image on order details without watermark

#### Expected behavior

images generated by img/tmp must be the same as the images produced (small-default or home_default)

#### Steps to Reproduce

1. Go to 'detail product'
2. see a image product with watermark
3. Go to 'order detail'
4. See a image product -> Not watermark
-----------------------
>Edit by Khouloud

Steps to reproduce the issue:
1. Go to the BO => Create a product `Product1` and upload an image `Image1`
2. Go to the FO => Add the last product created to the cart and process to checkout
3. Go to the BO => Orders => Last Order view page
4. Check the Product image => OK
5. Go to the BO => edit the product `Product1`
6. Remove the image `Image1`
7. Upload another image `Image2`
8. Save
9. Go to the BO=> Design => Image Settings page => Try to "Regenerate thumbnails"
10. Go to the BO => Orders => Last Order view page
11. Check the image of the product, it is the previous one `Image1` and not `Image2`.

**Screenshots**
In page product details:
![Capture_2](https://user-images.githubusercontent.com/61840433/76075090-af5f0b80-5f9c-11ea-9ef7-82b2a3ee0f35.JPG)

---

On page Order details:
![Capture_1](https://user-images.githubusercontent.com/61840433/76075134-c0a81800-5f9c-11ea-82cd-57c4283b3853.JPG)

#### Additional information

* PrestaShop version: 1.7.5.2
* PHP version: 7.0.33

**Comment by khouloudbelguith** [comment]
Hi @YoniSaadoun,

I have the same issue with Ps1763 & PS1764 and watermark v2.0.0.
![IMG_20200306_123945](https://user-images.githubusercontent.com/16067358/76079723-973fba00-5fa5-11ea-8b86-20c9d5def506.jpg)
But I think, it is an improvement instead of a bug.
The Type of the image in the order view details is product_mini
![product_mini_2_9](https://user-images.githubusercontent.com/16067358/76079823-cce4a300-5fa5-11ea-9d29-69b13bde8af8.jpg)
But in the module watermark configuration,this type doesn't exist 
![IMG_20200306_123407](https://user-images.githubusercontent.com/16067358/76079941-1a611000-5fa6-11ea-8eb5-5f2c763d1f55.jpg)
Ping @PrestaShop/prestashop-product-team what do you think?

Thanks!

**Comment by marionf** [comment]
Hello @YoniSaadoun 

The purpose of watermark module is to protect images in the front-office, so images can't be downloaded and used by someone.
As this is a request to improve the module, we need to know why you (or the user) want to do this and what is the interest ? 
This will help us to decide if we will implement this feature request or not

**Comment by YoniSaadoun** [comment]
The watermark its not a probleme for the customer. he will work without this watermark, its ok for him.

however, if a customer changes the cover photo of a product, the thumbnail of this product is not changed on the order page.
The thumbnail on the order page comes from the img/tmp folder. If you delete this thumbnail, the correct thumbnail will be well regenerated. But the question is, why are these temporary thumbnails not deleted and regenerated when you change an image in a product in bo, or when you empty the cache?

**Comment by YoniSaadoun** [comment]
When changing the coverage image, do the old commands containing this product keep the old image in memory? the next orders will have the new image.

If I want to get the good image for the old commands, you have to empty img/tmp? not auto generate.

**Comment by khouloudbelguith** [comment]
@YoniSaadoun, thanks for these clarifications.
This issue is not related to module watermark.
Steps to reproduce the issue:
1. Go to the BO => Create a product `Product1` and upload an image `Image1`
2. Go to the FO => Add the last product created to the cart and process to checkout
3. Go to the BO => Orders => Last Order view page
4. Check the Product image => OK
5. Go to the BO => edit the product `Product1`
6. Remove the image `Image1`
7. Upload another image `Image2`
8. Save
9. Go to the BO=> Design => Image Settings page => Try to "Regenerate thumbnails"
10. Go to the BO => Orders => Last Order view page
11. Check the image of the product, it is the previous one `Image1` and not `Image2`.

I have the same behavior as PS1.7.5.2 and PS1.7.6.3.
Order view page
![image](https://user-images.githubusercontent.com/16067358/76101073-1ba63300-5fce-11ea-8477-ae37292670d3.png)
Product details page
![image](https://user-images.githubusercontent.com/16067358/76101102-282a8b80-5fce-11ea-8516-bcd8c27e4492.png)
As I said before,
I think it is an improvement instead of a bug.
The Type of the image in the order view details is product_mini
![image](https://user-images.githubusercontent.com/16067358/76101232-632cbf00-5fce-11ea-92cc-7049f09bd739.png)
But, in the BO=> Design => Image Settings page this type doesn't exist
![image](https://user-images.githubusercontent.com/16067358/76101440-ba329400-5fce-11ea-82b2-9b475c0ca3ac.png)

Ping @MatShir @PrestaShop/prestashop-product-team what do you think? can we add a new type for orders?

Thanks!

**Comment by neich78** [comment]
Have the same issue (PS 1.7.6.4) and the client got confused with the order image does not match the right product image. 
This is order image which is under tmp/img directory as "mini"
![Captura de pantalla 2020-05-30 a las 11 50 16](https://user-images.githubusercontent.com/11721870/83325831-a914be80-a26f-11ea-9454-6d359b72e5b2.png)
This is product catalog image, also in tmp/img but with another name:
![Captura de pantalla 2020-05-30 a las 11 51 51](https://user-images.githubusercontent.com/11721870/83325833-aade8200-a26f-11ea-822d-109e5da3d446.png)

I have uploaded and replaced that image via FTP but this will happen again. Caché cleaning through the Backoffice does not clean that tmp directory and regenerate thumbnails neither. 
Any solution?

**Comment by ProteanCode** [comment]
still exists in 1.7.7.1

**Comment by davidegaretto** [comment]
Hi, was it fixed with the new version 1.7.8?

**Comment by nzebikes** [comment]
> Hi, was it fixed with the new version 1.7.8?

Actually I think it got worse: I upgraded from 1.7.7.x to latest 1.7.8.x and now even in the shopping cart it will show the wrong picture*! Since this is not acceptable for my store (and I guess for any store) and reverted back to the previous version. Pretty disappointing...
* to explain better: imagine you have 5 product options, each option is assigned a specific picture, when selecting the option in the front end it does show the correct picture to the customer, but when adding to the cart it will show always the picture of the 'first' product selection, but not the one the customer had selected, although the product code is correct.

**Comment by PrestaSafe** [comment]
I've the same issue for one of my client. 

To fix it, i've renamed the `img/tmp` by `img/tmp-old` and create a new  folder  `img/tmp` 
and it works fine !

Because if the image exist in the tmp folder, it won't be generated anymore.

**Comment by MatShir** [comment]
Hi @PrestaSafe,
Don't hesitate to propose a pull request ([find more information here](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/)). We will be happy to review your contribution to the PrestaShop core ;) <3

**Comment by alfonsolr09** [comment]
same problem on 1.7.8.5
How can I regenerate the images in the img/temp folder?

**Comment by PrestaSafe** [comment]
@alfonsolr09 just rename it ;)

**Comment by kpodemski** [comment]
Hello everyone. Following our new approach to feature requests, I converted this Issue to the GitHub Discussion so that the community can discuss the best possible ways to handle this problem and maybe, I hope, organize together to work towards fixing it in the core. If you are motivated to do so, we will be happy to help.

**Comment by ocio87** [comment]
I have the same problem.

The order shows an incorrect miniature image of the purchased product. It is solved by removing the jpg files from the img/tmp folder but it is not a root solution since it is presented again with other products.

Versión prestashop 8.1.1
Php: 8.1
